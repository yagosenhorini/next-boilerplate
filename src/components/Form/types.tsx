import { ReactNode } from 'react';

export type IForm = {
  children: ReactNode;
  handleSubmit: () => void;
};
