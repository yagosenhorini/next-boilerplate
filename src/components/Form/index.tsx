import * as S from './styled';

import { IForm } from './types';

const Form = ({ children, handleSubmit }: IForm) => (
  <S.Form data-testid="form" noValidate onSubmit={handleSubmit}>
    {children}
  </S.Form>
);

export default Form;
