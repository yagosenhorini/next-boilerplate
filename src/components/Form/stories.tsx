import { Meta, StoryObj } from '@storybook/react';
import Form from '.';

export default {
  title: 'Form',
  component: Form,
  tags: ['autodocs'],
} as Meta;

export const Default: StoryObj = {};
