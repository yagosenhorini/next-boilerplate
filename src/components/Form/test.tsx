import { render, screen } from '@testing-library/react';

import StyledProvider from '../../lib/theme';

import Form from '.';

describe('<Form />', () => {
  const mock = jest.fn();
  it('should render the component', () => {
    const { container } = render(
      <StyledProvider>
        <Form handleSubmit={mock}>
          <input type="text" />
        </Form>
      </StyledProvider>
    );

    expect(screen.getByTestId('form')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
