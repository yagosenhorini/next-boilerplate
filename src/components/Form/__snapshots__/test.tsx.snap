// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render the component 1`] = `
.c0 {
  width: 70%;
  display: flex;
  margin: 0 auto;
  align-items: center;
  flex-direction: column;
  justify-content: center;
}

<form
  class="c0"
  data-testid="form"
  novalidate=""
>
  <input
    type="text"
  />
</form>
`;
